lib/network/httputils/pagequery.go

```diff
@@ -0,0 +1,109 @@
+package httputils
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/node/runner/api/resource"
+	"boscoin.io/sebak/lib/storage"
+)
+
+const DefaultMaxLimit uint64 = 100
+
+type PageQuery struct {
+	request *http.Request
+	cursor  []byte
+	reverse bool
+	limit   uint64
+}
+
+func NewPageQuery(r *http.Request) (*PageQuery, error) {
+	p := &PageQuery{
+		request: r,
+		limit:   DefaultMaxLimit,
+	}
+	err := p.parseRequest()
+	return p, err
+}
+
+func (p *PageQuery) Limit() uint64 {
+	return p.limit
+}
+
+func (p *PageQuery) Reverse() bool {
+	return p.reverse
+}
+
+func (p *PageQuery) Cursor() []byte {
+	return p.cursor
+}
+
+func (p *PageQuery) SelfLink() string {
+	return p.request.URL.String()
+}
+
+func (p *PageQuery) PrevLink(cursor []byte) string {
+	path := p.request.URL.Path
+	query := p.urlValues(cursor, true, p.limit).Encode()
+	link := fmt.Sprintf("%s?%s", path, query)
+	return link
+}
+
+func (p *PageQuery) NextLink(cursor []byte) string {
+	path := p.request.URL.Path
+	query := p.urlValues(cursor, false, p.limit).Encode()
+	link := fmt.Sprintf("%s?%s", path, query)
+	return link
+}
+
+func (p *PageQuery) ListOptions() storage.ListOptions {
+	return storage.NewDefaultListOptions(p.Reverse(), p.Cursor(), p.Limit())
+}
+
+func (p *PageQuery) ResourceList(rs []resource.Resource, cursor []byte) *resource.ResourceList {
```

Comment:
IMO, `httputils` package does not need to depend on the package `resource`
Why don't you call `resource.NewResourceList` at the location where call this function? `ResourceList` ?

Reply:
It's a good point. If `resource` pkg uses `httputils`, the `PageQuery` makes circular import.. huum..